integrated_channels/blackboard/models.py

```diff
@@ -88,3 +94,93 @@ def get_content_metadata_exporter(self, user):
 
     def get_content_metadata_transmitter(self):
         return BlackboardContentMetadataTransmitter(self)
+
+    def get_learner_data_exporter(self, user):
+        return BlackboardLearnerExporter(user, self)
+
+    def get_learner_data_transmitter(self):
+        return BlackboardLearnerTransmitter(self)
+
+
+@python_2_unicode_compatible
+class BlackboardLearnerDataTransmissionAudit(models.Model):
+    """
+    The payload we send to Blackboard at a given point in time for an enterprise course enrollment.
+
+    """
+    blackboard_user_email = models.EmailField(
+        max_length=255,
+        blank=False,
+        null=False,
+        help_text='The learner`s Blackboard email. This must match the email on edX'
+    )
+
+    enterprise_course_enrollment_id = models.PositiveIntegerField(blank=False, null=False, db_index=True)
+    course_id = models.CharField(max_length=255, blank=False, null=False)
+    course_completed = models.BooleanField(default=False)
+    grade = models.DecimalField(blank=True, null=True, max_digits=3, decimal_places=2)
+    total_hours = models.FloatField(null=True, blank=True)
+    course_completed = models.BooleanField(
+        default=True,
+        help_text="The learner's course completion status transmitted to Blackboard."
+    )
+    completed_timestamp = models.CharField(
+        max_length=10,
+        help_text=(
+            'Represents the Blackboard representation of a timestamp: yyyy-mm-dd, '
+            'which is always 10 characters.'
+        )
+    )
+
+    # Request-related information.
+    status = models.CharField(max_length=100, blank=False, null=False)
+    error_message = models.TextField(blank=True)
+    created = models.DateTimeField(auto_now_add=True)
+
+    class Meta:
+        app_label = 'blackboard'
+
+    def __str__(self):
+        """
+        Return a human-readable string representation of the object.
+        """
+        return (
+            '<BlackboardLearnerDataTransmissionAudit {transmission_id} for enterprise enrollment '
+            '{enterprise_course_enrollment_id}, Blackboard user {blackboard_user_email}, '
+            'and course {course_id}>'.format(
+                transmission_id=self.id,
+                enterprise_course_enrollment_id=self.enterprise_course_enrollment_id,
+                blackboard_user_email=self.blackboard_user_email,
+                course_id=self.course_id
+            )
+        )
+
+    def __repr__(self):
+        """
+        Return uniquely identifying string representation.
+        """
+        return self.__str__()
+
+    def serialize(self, *args, **kwargs):  # pylint: disable=unused-argument
+        """
+        Return a JSON-serialized representation.
+
+        Sort the keys so the result is consistent and testable.
+
+        # TODO: When we refactor to use a serialization flow consistent with how course metadata
```

Comment:
interesting, curious to learn more here but okay if this works

Reply:
This was including from other channels before we started our integrations. Not sure if it's still in effect?

Reply:
yeah we will want to check this out